modules/andes-core/client/src/main/java/org/wso2/andes/client/XASession_9_1.java

```diff
@@ -154,7 +157,10 @@ XaStatus startDtx(Xid xid, int flag) throws FailoverException, AMQException {
      * @throws FailoverException when a connection issue is detected
      * @throws AMQException      when an error is detected in AMQ state manager
      */
-    public XaStatus endDtx(Xid xid, int flag) throws FailoverException, AMQException {
+    public XaStatus endDtx(Xid xid, int flag) throws FailoverException, AMQException, XAException {
+
+        throwErrorIfClosed();
+
         if (LOGGER.isDebugEnabled()) {
```

Comment:
I don't think so since we are not sending the dtx.end if the session is closed. If we move it up we can have a log entry which says "Sending dtx.end for channel ..." but actually we never tried doing that.